web/react/stores/post_store.jsx

```diff
@@ -99,8 +99,57 @@ var PostStore = assign({}, EventEmitter.prototype, {
         }
         return null;
     },
-    storePosts: function storePosts(channelId, posts) {
-        this.pStorePosts(channelId, posts);
+    storePosts: function storePosts(channelId, newPostList) {
+        if (isPostListNull(newPostList)) {
+            return;
+        }
+
+        var postList = PostStore.getPosts(channelId);
+        if (isPostListNull(postList)) {
+            postList = newPostList;
+        } else {
+            postList = makePostListNonNull(postList);
```

Comment:
Little bit of paranoia here? We check if it is not null then really make sure of it!